build.sc

```diff
@@ -42,18 +42,22 @@ trait MainArgsPublishModule
     super.scalacOptions() ++
       Option.when(!isScala3(scalaVersion()))("-P:acyclic:force")
 
-  def scalacPluginIvyDeps =
+  def scalacPluginIvyDeps = T{
     super.scalacPluginIvyDeps() ++
-      Option.when(!isScala3(scalaVersion()))(ivy"com.lihaoyi:::acyclic:${acyclic}")
+    Agg(ivy"com.lihaoyi::unroll-plugin:0.1.9") ++
+    Option.when(!isScala3(scalaVersion()))(ivy"com.lihaoyi:::acyclic:${acyclic}")
+  }
 
   def compileIvyDeps =
     super.compileIvyDeps() ++
-      Agg.when(!isScala3(crossScalaVersion))(
-        ivy"com.lihaoyi:::acyclic:${acyclic}",
-        ivy"org.scala-lang:scala-reflect:$crossScalaVersion"
-      )
+    Agg.when(!isScala3(crossScalaVersion))(
+      ivy"com.lihaoyi:::acyclic:${acyclic}",
+      ivy"org.scala-lang:scala-reflect:$crossScalaVersion",
+
+    )
 
   def ivyDeps = Agg(
+    ivy"com.lihaoyi::unroll-annotation:0.1.9",
```

Comment:
Is `unroll-annotation` needed transitively, or only at compile time?

Reply:
Only at compile time. I can have the plugin remove the annotation if necessary, haven't thought deeply about whether that's the right thing to do

Reply:
We should move it into `compileIvyDeps` then.

Since we are more or less generating stuff that is not part of the actual API but rather some inconvenient necessity, not forcing a transitive dependency on this implementation detail for all downstream users seem to be the right thing.